modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/graphql/dto/v1_0/ObjectDefinitionGraphQLDTOContributor.java

```diff
@@ -260,17 +260,43 @@ public <T> T getRelationshipValue(
 		ObjectEntry objectEntry = _objectEntryManager.getObjectEntry(
 			dtoConverterContext, _objectDefinition, id);
 
-		long relationshipId = _getRelationshipId(objectEntry.getProperties());
+		Map<String, Object> properties = objectEntry.getProperties();
+
+		long relationshipId = _getRelationshipId(properties);
 
 		if (relationshipId <= 0) {
-			Page<ObjectEntry> page =
-				_objectEntryManager.getObjectEntryRelatedObjectEntries(
-					dtoConverterContext, _objectDefinition, id,
-					relationshipName,
-					Pagination.of(QueryUtil.ALL_POS, QueryUtil.ALL_POS));
-
-			return (T)TransformUtil.transform(
-				page.getItems(), itemObjectEntry -> _toMap(itemObjectEntry));
+			String objectField2Name = _getObjectField2Name(
```

Comment:
hey @DaneszLR, could you please move [this](https://github.com/liferay-objects/liferay-portal/pull/1956/files#diff-e97849f7166af058a273ab29d56d27e04f345ed06e9f42db2b26cbfed5df0751R268-R299) logic out of the if?

Reply:
Hey @seltong, I can't really move it out of the if, because _getRelationshipId(properties) checks for a match with a specific pattern. This method was made for a Many-to-Many relationship error. In One-to-Many relationships that pattern does not appear (I tested it multiple times, but please correct me if I'm wrong). So relationshipId will always be 0 when it comes to one-to-many realtionships. 

But if move my logic above this "if (relationshipId <= 0)" check then the program would never reach that point.

Reply:
Hi, @DaneszLR. I believe that the relationshipId should never be 0 or less than 0.

The method `_getRelationshipId` looks inconsistent for me because it is returning the first long value found by a key that follow the relationship name pattern.
If we create more than one relationship, the first relationship will always be returned.

How reproduce this wrong case:

> Create a 1:M relationship User to GraphqlTest
> Create a 1:M relationship Custom Object -> GraphqlTest
> Try to get the first relationship by the graphQL api
> Try to get the second relationship by the graphQL api

In addiction, the name of the variable `relationshipId` doesn't seem good to me. By my investigation, the value assigned to this variable is the `objectEntryId` of the related ObjectEntry and not the `relationshipId`.

For example: in the case of the  relationship created on the steps of [LPS-178473](https://issues.liferay.com/browse/LPS-178473), a property `r_graphqlTestUser_userId` will be inside the `objectEntry.getProperties()`. The value associated with that property is the `userId` of the User selected during the creation of the object entry in step 7.

Please let me know if I'm missing something or it doesn't make sense, but the method `_getRelationshipId` looks inconsistent for me.

Could you please analyse the comments above and confirm if it makes sense?

Thanks in advance

Reply:
Hi @seltong,

The method `_getRelationshipId` is not my work. But I see your error. Trying to merge my solution with that method. Changing the pattern and checking for relationshipName should do the trick. That method came in on the same day the first PR was put up. From the name of the commit, it fixes a similar issue that I also had (Key names in map ending with ERC while also having the same relationshipName so the method returned a wrong value, not the Long type ), but it wasn't checked for keys that doesn't match the pattern, but are valid keys.

Fixing and testing. Will update you once I'm done with a solution.
Dani

Reply:
Hi @seltong. I think I managed to do it. I'm building it right now then I will test it. After that I will open a new PR so there will be no conflicts or unorganized commits.